Add option for generic Python tests

vepadulano opened the pull request:
Follow up to https://github.com/root-project/root/pull/17950

In the linked PR we found ourselves in need to create a Python ctest test that uses `pytest` as the test driver. The currently available `ROOT_ADD_PYUNITTEST` cmake function is limited because it only allows Python tests written with `unittest`, so we can introduce a new one more generic.

dpiparo reviewed (approved):
Thanks for this work. For me the PR can be merged as is, nevertheless let me ask a question: have you considered perhaps unifying the two functions not to add another one to the CMake infra?

vepadulano commented:
>  have you considered perhaps unifying the two functions not to add another one to the CMake infra?

Thanks for the idea, I implemented and rebased :+1:
